Purchases/Misc/Atomic.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Atomic.swift
+//
+//  Created by Nacho Soto on 11/20/21.
+
+import Foundation
+
+/// An atomic variable.
+internal final class Atomic<T> {
```

Comment:
nitpick: I kinda feel like if we improve the class description here, the comments on each operation could be removed. 
like, if we had something like: 
```swift
/** A generic object that performs all write and read operations atomically. 
* Use it to prevent data races when accessing an object. 
* 
* Usage: 
* ```swift
* let foo = Atomic<MyClass>
* // read values
* foo.withValue { 
*     currentBar = $0.bar
*     currentX = $0.x
* }
* // write value
* foo.modify { 
*     $0.bar = 2
*     $0.x = "new X"
* }
* ```
* 
* Or for single-line read/writes: 
* ```swift
* let currentX = foo.value.x
* foo.value = MyClass()
* ``` 
**/
```

Then the usage and even parts of the implementation might seem clearer on first glance, and the other comments would be redundant, since we know all operations are performed atomically, from the class name and the docs

Reply:
Oh that's a great idea! Much easier to understand than poorly worded explanations of each method 😛

Reply:
Oh that's a great idea! Much easier to understand than poorly worded explanations of each method 😛